lib/tasks/generate-from-blueprint.js

```diff
@@ -23,7 +23,7 @@ class GenerateTask extends Task {
     let addonBlueprint = this.lookupBlueprint(`${name}-addon`, true);
     // otherwise, use default addon-import
 
-    if (noAddonBlueprint.indexOf(name) < 0 && !addonBlueprint && (mainBlueprint && mainBlueprint.supportsAddon()) && options.args[1]) {
+    if (noAddonBlueprint.indexOf(name) < 0 && !addonBlueprint && (mainBlueprint && mainBlueprint.supportsAddon(options)) && options.args[1]) {
```

Comment:
This is a pretty complex statement. For clarity, consider extracting all or parts of this to intention revealing variables.
```javascript
const addonBlueprintNotFound = noAddonBlueprint.indexOf(name) < 0;
const addonSupported = mainBlueprint && mainBlueprint.supportsAddon(options);
const blueprintName = options.args[1]; // just a guess
if (!addonBlueprint && addonBlueprintNotFound && addonSupported  && blueprintName) {
```